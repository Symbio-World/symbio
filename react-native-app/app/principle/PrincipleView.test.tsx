import * as React from 'react'
import { render, fireEvent } from 'react-native-testing-library'
import { TouchableOpacity } from 'react-native'
import { PrincipleView } from './PrincipleView'

describe('PrincipleView', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<PrincipleView title="Test" />)
    expect(toJSON()).toMatchSnapshot()
  })

  it('can be selected', () => {
    const { toJSON } = render(<PrincipleView title="Test" selected />)

    expect(toJSON()).toMatchSnapshot()
  })

  it('calls onPress', () => {
    const handlePress = jest.fn()
    const { getByType } = render(
      <PrincipleView title="Test" onPress={handlePress} />,
    )
    fireEvent.press(getByType(TouchableOpacity))
    expect(handlePress).toHaveBeenCalled()
  })
})
