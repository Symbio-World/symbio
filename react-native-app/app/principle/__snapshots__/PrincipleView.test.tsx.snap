// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrincipleView can be selected 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#48bb78",
      "borderColor": "#48bb78",
      "borderRadius": 9999,
      "borderWidth": 2,
      "margin": 12,
      "opacity": 1,
      "paddingHorizontal": 24,
      "paddingVertical": 12,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 20,
        },
        Object {
          "color": "#fff",
        },
        Object {
          "fontWeight": "bold",
        },
      ]
    }
  >
    Test
  </Text>
</View>
`;

exports[`PrincipleView renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#fff",
      "borderColor": "#48bb78",
      "borderRadius": 9999,
      "borderWidth": 2,
      "margin": 12,
      "opacity": 1,
      "paddingHorizontal": 24,
      "paddingVertical": 12,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 20,
        },
        Object {
          "color": "#48bb78",
        },
        Object {
          "fontWeight": "bold",
        },
      ]
    }
  >
    Test
  </Text>
</View>
`;
